src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1036,10 +1039,33 @@ impl Coordinator {
                 cluster_id,
                 non_null_assertions,
                 custom_logical_compaction_window: compaction_window,
+                refresh_schedule: refresh_schedule.clone(),
             }),
             owner_id: *session.current_role_id(),
         });
 
+        // Timestamp selection
+        let as_of = {
+            // Normally, `as_of` should be the least_valid_read.
+            let id_bundle = dataflow_import_id_bundle(global_lir_plan.df_desc(), cluster_id);
+            let mut as_of = self.least_valid_read(&id_bundle);
+            // But for MVs with non-trivial REFRESH schedules, it's important to set the
+            // `as_of` to the first refresh. This is because we'd like queries on the MV to
+            // block until the first refresh (rather than to show an empty MV).
+            if let Some(refresh_schedule) = refresh_schedule {
+                if let Some(ts) = as_of.as_option() {
+                    let Some(rounded_up_ts) = refresh_schedule.round_up_timestamp(*ts) else {
+                        return Err(AdapterError::MaterializedViewWouldNeverRefresh(name.item));
+                    };
+                    as_of = Antichain::from_elem(rounded_up_ts);
```

Comment:
I suggest to add some a structured log line with some message and `rounded_up_ts` and `refresh_schedule` reported as key-value pairs.